[mlir][spirv] Make ConvertToSPIRVPass into a test pass (non-public)

andfau-amd opened the pull request:
With the removal of mlir-vulkan-runner (as part of #73457) in e7e3c45bc70904e24e2b3221ac8521e67eb84668, this pass no longer has to be public (previously it had to be so the runner could use it). This commit makes it instead only available for use by mlir-opt.

kuhar commented on mlir/test/lib/Pass/TestConvertToSPIRVPass.cpp:
```diff
@@ -71,9 +66,44 @@ void populateConvertToSPIRVPatterns(const SPIRVTypeConverter &typeConverter,
+  }
+
+  TestConvertToSPIRVPass() = default;
+  TestConvertToSPIRVPass(bool convertGPUModules, bool nestInGPUModule) {
+    this->convertGPUModules = std::move(convertGPUModules);
+    this->nestInGPUModule = std::move(nestInGPUModule);
```
There's no point in calling `move` on bools. Also, prefer initialization lists in constructors.

andfau-amd replied:
These can't be set with the initialization list, you have to use the assign operator as far as I can tell.

kuhar replied:
Why is that? I'd expect this to work:
```suggestion
  TestConvertToSPIRVPass(bool convertGPUModules, bool nestInGPUModule)
    : convertGPUModules(convertGPUModules), nestInGPUModule(nestInGPUModule) {
```

kuhar replied:
Oh, I understand -- these are initialized first and then set to a specific value. Nerve mind then.

kuhar replied:
Maybe remove the moves only

andfau-amd replied:
I tried that! But the constructor for `Option` is only for declaring the option's name and description and so on; there isn't a constructor for instead initializing it with just a value (because then it'd have no name and description, maybe?).

andfau-amd replied:
Basically, it has a weird dual role of holding the value of the option and holding a description of the option, and this is the place where it trips you up.

kuhar commented on mlir/test/lib/Pass/TestConvertToSPIRVPass.cpp:
```diff
@@ -71,9 +66,44 @@ void populateConvertToSPIRVPatterns(const SPIRVTypeConverter &typeConverter,
+    this->convertGPUModules = std::move(convertGPUModules);
+    this->nestInGPUModule = std::move(nestInGPUModule);
+  };
+  TestConvertToSPIRVPass(const TestConvertToSPIRVPass &) {}
```
```suggestion
  TestConvertToSPIRVPass(const TestConvertToSPIRVPass &) = default;
```
to be consistent with the other constructor above

andfau-amd replied:
That constructor gets implicitly deleted.

kuhar commented on mlir/test/lib/Pass/TestConvertToSPIRVPass.cpp:
```diff
@@ -134,3 +164,16 @@ struct ConvertToSPIRVPass final
 
 } // namespace
+
+namespace mlir {
+namespace test {
```
```suggestion
namespace mlir::test {
```

kuhar commented on mlir/test/lib/Pass/TestVulkanRunnerPipeline.cpp:
```diff
@@ -26,6 +25,14 @@
 using namespace mlir;
 
+// Defined in the test directory, no public header.
+namespace mlir {
+namespace test {
```
```suggestion
namespace mlir::test {
```

kuhar reviewed (commented):
Looks good, just a few nits

andfau-amd committed:
[mlir][spirv] Make ConvertToSPIRVPass into a test pass (non-public)

With the removal of mlir-vulkan-runner (as part of #73457) in
e7e3c45bc70904e24e2b3221ac8521e67eb84668, this pass no longer has to be
public (previously it had to be so the runner could use it). This commit
makes it instead only available for use by mlir-opt.

andfau-amd commented:
Reverted in 4573c857da88b3210d497d9a88a89351a74b5964, I must have missed some dependencies in the CMakeLists.

andfau-amd commented:
Recommitted as 25ae1a266d50f24a8fffc57152d7f3c3fcb65517.
